### ISSUE TITLE
Add support for binary classification

### ISSUE THREAD

**Issue by LittleLittleCloud** [dotnet/machinelearning-modelbuilder#1344]
We have a few users ask for binary classifications. Considering that using binary trainers can be faster compared to the solution of transferring all binary into multi and use multi-trainers instead, and the generated pipeline will be more suitable for editing, it might be time to let binary classification back.

One major blocking issue for adding binary classification is the uncertainty of the label format. It will be confusing if the input label is '0/1' while the output result is 'false/true'. However, this can be solved by firstly mapping input label to key, and then convert the key to boolean type. And after the trainer transforms input and writes output label to "PredictedLabel",  we can map the boolean type "PredictedLabel" back to original label by firstly converting it to key type and feed it into "MapKeyToValue". The converting pipeline will be similar to this

`MapValueToKey("label") => ConvertType("label", boolean) => binaryTrainer("label", "PredictedLabel") => ConvertType("PredictedLabel", int32) => MapKeyToValue("PredictedLabel", "PredictedLabel")`

**Comment by LittleLittleCloud** [comment]
#1211
